src/osd/scrubber_common.h

```diff
@@ -24,6 +25,21 @@ namespace Scrub {
   struct ReplicaActive;
 }
 
+/// reservation-related data sent by the primary to the replicas,
+/// and used to match the responses to the requests
+struct AsyncScrubResData {
+  spg_t pgid;
+  pg_shard_t from;
+  epoch_t request_epoch;
+  MOSDScrubReserve::reservation_nonce_t nonce;
```

Comment:
Initialize nonce and request_epoch to 0 or disallow the default constructor and ensure that all constructors initialize them to a parameter.  We don't want it to be possible to construct an AsyncScrubResData with uninitialized fields.